mlir/lib/Dialect/Rock/Transforms/MakeGepsInbounds.cpp

```diff
@@ -0,0 +1,64 @@
+//===- CleanMath.cpp - Clean up math after lowering/unrolling loops  ---===//
+//
+// Copyright 2022 AMD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Arith/Transforms/Passes.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Rock/Passes.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/Passes.h"
+
+// Remove these includes after we're not working with a test pass upstream
+#include "mlir/Analysis/DataFlow/DeadCodeAnalysis.h"
+#include "mlir/Analysis/DataFlow/IntegerRangeAnalysis.h"
+#include "mlir/Transforms/FoldUtils.h"
+
+
+namespace mlir {
+namespace rock {
+// TODO: ask why?
+#define GEN_PASS_DEF_ROCKMAKEGEPSINBOUNDSPASS
+#include "mlir/Dialect/Rock/Passes.h.inc"
+} // namespace rock
+} // namespace mlir
+
+#define DEBUG_TYPE "rock-make-geps-inbounds"
+
+using namespace mlir;
+// Remove after upstream integrates
+using namespace mlir::dataflow;
+
+namespace {
+struct RockMakeGepsInboundsPass
+    : public rock::impl::RockMakeGepsInboundsPassBase<RockMakeGepsInboundsPass> {
+  void runOnOperation() override;
+};
+} // end namespace
+
+
+void RockMakeGepsInboundsPass::runOnOperation() {
+  auto func = getOperation();
+  llvm::outs() << func->getName() << ": attrs: ";
+  for (auto attr: func->getAttrs()) {
+    llvm::outs() << ", " << attr.getName();
+  }
+  llvm::outs() << '\n';
+
+  func.walk([](LLVM::GEPOp gepOp) {
+    llvm::outs() << gepOp << '\n';
```

Comment:
... ah, right, this is a draft.

Here, you'll probably just want to add the `inbounds` attribute (I forget if it's a boolean or a `UnitAttr`)

Reply:
yes, it is a draft. I just decided to share with you and @manupak. I mean I just wanted to be sure that I am moving to a correct direction.

Yes, I will just add `inbounds` attribute then.